pkg/vm/vm.go

```diff
@@ -1104,7 +1104,10 @@ func (v *VM) execute(ctx *Context, op opcode.Opcode, parameter []byte) (err erro
 		arr := elem.Bytes()
 		v.estack.PushVal(len(arr))
 
-	case opcode.JMP, opcode.JMPIF, opcode.JMPIFNOT:
+	case opcode.JMP:
+		v.jump(ctx, parameter)
+
+	case opcode.JMPIF, opcode.JMPIFNOT:
```

Comment:
There is still some code duplication left, it'd be nice to unify parameter parsing and checking.